instrumentation/r2dbc-1.0/library/src/main/java/io/opentelemetry/instrumentation/r2dbc/v1_0/R2dbcTelemetryBuilder.java

```diff
@@ -39,6 +39,16 @@ public R2dbcTelemetryBuilder setStatementSanitizationEnabled(boolean enabled) {
     return this;
   }
 
+  /**
+   * Overrides the version of the instrumentation. If not set, the version is automatically
+   * detected.
+   */
+  @CanIgnoreReturnValue
+  public R2dbcTelemetryBuilder setInstrumentationVersion(String instrumentationVersion) {
```

Comment:
why do you need this? why doesn't the usual instrumentation version properties file work for you?

Reply:
you can probably fix it by modifying https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/eb8748b65b17a5d8efde381bc5294247a6b2134d/instrumentation/r2dbc-1.0/library-instrumentation-shaded/build.gradle.kts#L15 so that it would not exclude the instrumentation version properties

Reply:
I've tried `exclude("META-INF/services/*")` but it wouldn't exclude `META-INF/services/io.r2dbc.spi.ConnectionFactoryProvider`

@laurit do you have an idea?

Reply:
@laurit thanks a lot for the help with this PR :100: